src/Reflection/ReflectionClass.php

```diff
@@ -1287,36 +1289,28 @@ public function __clone()
 
     /**
      * Get the value of a static property, if it exists. Throws a
-     * PropertyDoesNotExist exception if it does not exist or is not static.
+     * PropertyDoesNotExistOrIsNotStatic exception if it does not exist or is not static.
      * (note, differs very slightly from internal reflection behaviour)
      *
      * @param string $propertyName
      *
      * @return mixed
      *
-     * @throws PropertyDoesNotExist
+     * @throws PropertyNotAccessible
      * @throws ClassDoesNotExist
+     * @throws NoObjectProvided
+     * @throws NotAnObject
+     * @throws ObjectNotInstanceOfClass
      */
     public function getStaticPropertyValue(string $propertyName)
     {
-        if ( ! \class_exists($this->getName(), false)) {
-            throw new ClassDoesNotExist('Property cannot be retrieved as the class is not loaded');
-        }
-
         $property = $this->getProperty($propertyName);
 
         if ( ! $property || ! $property->isStatic()) {
-            throw new PropertyDoesNotExist('Property does not exist on class or is not static');
-        }
-
-        // PHP behaviour is to simply say "property does not exist" if accessing
-        // protected or private values. Here we be a little more explicit in
-        // reasoning...
-        if ( ! $property->isPublic()) {
-            throw new PropertyNotPublic('Property is not public');
+            throw PropertyDoesNotExistOrIsNotStatic::fromName($propertyName);
```

Comment:
You know if it does not exist or if it is static, so throwing a `PropertyDoesNotExistOrIsNotStatic` feels wrong.

A better way to handle this is having `PropertyDoesNotExistOrIsNotStatic` be an abstract class, and then have two subtypes for the actual failure: that would make it much easier to deal with for a consumer, and would read more sensibly.

Reply:
Ok. I just used the code that was already here :) https://github.com/Roave/BetterReflection/blob/master/src/Reflection/ReflectionClass.php#L1309

Reply:
Aye - I know my comments sometimes sound annoying and out of context, but I really just see the diffs only ^_^

Reply:
It's ok :) I hope I will someday prepare a pull request that can be merged straight away :)